src/sidebar/components/ThreadCard.tsx

```diff
@@ -63,22 +61,30 @@ function ThreadCard({ frameSync, thread }: ThreadCardProps) {
   }, [focusRequest, store, thread.id]);
 
   return (
-    /* eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions */
     <Card
       active={isHovered}
       classes="cursor-pointer focus-visible-ring theme-clean:border-none"
       data-testid="thread-card"
       elementRef={cardRef}
-      tabIndex={-1}
+      tabIndex={0}
```

Comment:
Do you know why we were previously setting `tabIndex={-1}` here? Given that this was a non-interactive element, I don't believe it would have been in the tab sequence anyway. Setting the `-1` value insinuates that something was controlling the tab sequence dynamically?

A couple of tasks here:

* Can you dig up any context around why `tabIndex` was set here previously?
* Can you confirm that `tabIndex` needs to be set on an element with `role="button"` (`<button>` elements get an implied `tabIndex=0` and I'm wondering if `role="button"` elements do also)? We should only set `tabIndex` if it's necessary.

Reply:
> Can you confirm that tabIndex needs to be set on an element with role="button"

Yes, I can confirm this. Without a `tabIndex` the element is skipped from the tab sequence.

> Can you dig up any context around why tabIndex was set here previously

Sure! But I believe you are in the right track when saying `something was controlling the tab sequence dynamically`, because the card already had the focus-ring-related classes, so something was already in place with the ability to focus it.

Reply:
@lyzadanger the reason it had `tabIndex={-1}` before and that it already had the focus ring class, is this:

[Grabación de pantalla desde 2023-06-14 09-40-56.webm](https://github.com/hypothesis/client/assets/2719332/3290f97e-3665-4fe0-81f4-d55bd777716f)

When the guest detects there's an existing annotation in your currently selected text, we allow to "show" it in the sidebar, which in practice focuses it.

It works by doing a `'showAnnotations'` RPC call from guest to sidebar, and then the sidebar listens to that call and triggers a `SET_ANNOTATION_FOCUS_REQUEST` redux action.

Then, the ''ThreadCard` itself handles that like this:

```tsx
const focusRequest = store.annotationFocusRequest();
const cardRef = useRef<HTMLElement | null>(null);
useEffect(() => {
  if (focusRequest !== thread.id || !cardRef.current) {
    return;
  }
  cardRef.current.focus();
  store.clearAnnotationFocusRequest();
}, [focusRequest, store, thread.id]);

return (
  <Card
    elementRef={cardRef}
...
```

Reply:
OK, to confirm: `ThreadCard`s were previously not in the tab sequence but needed to be focus-able. After these changes they will be both in the tab sequence and focus-able.

Reply:
Yes, that's correct.